oem/ibm/libpldmresponder/file_io_type_smsmenu.cpp

```diff
@@ -0,0 +1,174 @@
+                      sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                          userPassLen);
+    std::string passStr(result.begin(), result.end());
+
+    if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_CHANGE)
+    {
+        // Split the vector to retrieve the New Password
+        result = vecSplit(smsBuf,
+                          sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                              userPassLen + sizeof(uint32_t),
+                          sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                              userPassLen + sizeof(uint32_t) + userNewPassLen);
+        std::string tempNewPassStr(result.begin(), result.end());
+        newPassStr = tempNewPassStr;
+    }
+
+    metaDataObj.fileStatus = 0x00;
+    metaDataObj.fileMetaData1 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData2 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData3 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData4 = 0xFFFFFFFF;
+
+    bool authenticated = false;
+    bool passwordChangeRequired = false;
+    bool operationAllowed = false;
+    enum changePasswordReasonCode changePassRc
+    {
+    };
+
+    if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_AUTHENTICATION)
+    {
+        authenticate(unameStr, passStr, authenticated, passwordChangeRequired,
+                     operationAllowed);
+        if (authenticated == true)
+        {
+            metaDataObj.fileMetaData1 = PasswordAuthResult::AUTHENTICATED;
+        }
+        else
+        {
+            metaDataObj.fileMetaData1 = PasswordAuthResult::NOT_AUTHENTICATED;
+        }
+
+        if (passwordChangeRequired == true)
+        {
+            metaDataObj.fileMetaData1 =
+                PasswordAuthResult::AUTHENTICATED_BUT_EXPIRED;
+        }
+
+        if (operationAllowed == true)
+        {
+            metaDataObj.fileMetaData2 = AdminOperationStatus::ALLOWED;
+        }
+    }
+    else if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_CHANGE)
+    {
+        changePassRc = changePassword(unameStr, passStr, newPassStr,
+                                      operationAllowed);
+        if (changePassRc == PASSWORDCHANGESUCCESSFUL)
+        {
+            metaDataObj.fileMetaData1 = PasswordChangeResult::CHANGED;
+        }
+        else if (changePassRc == NOTAUTHENTICATED)
+        {
+            metaDataObj.fileMetaData1 = PasswordChangeResult::NOT_CHANGED;
+        }
+        else
+        {
+            metaDataObj.fileMetaData1 = PasswordChangeResult::INVALID_PASSWORD;
+        }
+
+        if (operationAllowed == true)
+        {
+            metaDataObj.fileMetaData2 = AdminOperationStatus::ALLOWED;
+        }
+
+        metaDataObj.fileMetaData3 = (int)changePassRc;
+    }
+    info("SMS PAM authenticated={USR_AUTH} passwordChangeRequired={USR_PASS_CR}"
+         " operationAllowed={OPA}",
+         "USR_AUTH", authenticated, "USR_PASS_CR", passwordChangeRequired,
+         "OPA", operationAllowed);
+
+    return rc;
```

Comment:
directly return as success since rc is unassigned value from function

Reply:
Done